src/osp/link/machines.h

```diff
@@ -90,10 +83,14 @@ struct UpdMachPerType
     std::vector<BitVector_t> m_localDirty;
 };
 
-struct Junction
+struct MachinePair
 {
     MachLocalId     m_local{lgrn::id_null<MachLocalId>()};
     MachTypeId      m_type{lgrn::id_null<MachTypeId>()};
+};
+
+struct Junction : MachinePair
+{
     JuncCustom      m_custom{0};
 };
```

Comment:
MachinePair is 32bit + 16bit with an align of 32bit, so it has 16bit of padding.

Does Junction inherit from MachinePair in such a way that sizeof(Junction) == 64bit ? Or is it tacking on JuncCustom (16bit) after the padding from MachinePair, giving you a size of 96bit ?

I recommend a static assert on the sizeof(Junction) to make sure the compiler is behaving itself.

Reply:
In this case it's just easier to copy the members; this eliminates the problem.